### PR TITLE
LTD-5005 and LTD-5006 remove spire_id and reference from CSV upload

### PR DESCRIPTION
**Pull request by markj0hnst0n**
### Aim

To remove reference and spire_id fields from denial csv upload.

Corresponding Frontend PR: https://github.com/uktrade/lite-frontend/pull/1955

[LTD-5005](https://uktrade.atlassian.net/browse/LTD-5005)
[LTD-5006](https://uktrade.atlassian.net/browse/LTD-5006)

[LTD-5005]: https://uktrade.atlassian.net/browse/LTD-5005?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[LTD-5006]: https://uktrade.atlassian.net/browse/LTD-5006?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by depsiatwal — COMMENTED** [review]
LGTM is there a PR to update the front end example CSV ?

File: api/external_data/serializers.py [hunk 1]
```diff
@@ -40,14 +42,14 @@ class Meta:
...
     regime_reg_ref = serializers.CharField(source="denial.regime_reg_ref", required=False)
-    reference = serializers.CharField(source="denial.reference", required=False)
     item_list_codes = serializers.CharField(source="denial.item_list_codes", required=False)
     notifying_government = serializers.CharField(source="denial.notifying_government", required=False)
     item_description = serializers.CharField(source="denial.item_description", required=False)
     end_use = serializers.CharField(source="denial.end_use", required=False)
     is_revoked = serializers.BooleanField(source="denial.is_revoked", required=False)
     is_revoked_comment = serializers.CharField(source="denial.is_revoked_comment", required=False)
     reason_for_refusal = serializers.CharField(source="denial.reason_for_refusal", required=False)
+    reference = serializers.CharField(source="denial.reference", required=False, allow_blank=True)
```
Comment (currycoder) [inline api/external_data/serializers.py:52]: How come this variable is getting moved around?  Pretty minor, but if you can avoid moving things like this the PR is going to be easier to read and git history more understandable

File: api/external_data/tests/test_views.py [hunk 2]
```diff
@@ -273,11 +262,13 @@ def test_update_success(self):
...
         url = reverse("external_data:denial-list")
         content = """reference,regime_reg_ref,name,address,notifying_government,country,item_list_codes,item_description,end_use,reason_for_refusal,spire_entity_id,entity_type
-        ,AB-CD-EF-000,Organisation Name,"1000 Street Name, City Name",Country Name,Country Name,0A00100,Medium Size Widget,Used in industry,Risk of outcome,123,end_user
+        DN2000/0000,,Organisation Name,"1000 Street Name, City Name",Country Name,Country Name,0A00100,Medium Size Widget,Used in industry,Risk of outcome,123,end_user
         """
         response = self.client.post(url, {"csv_file": content}, **self.gov_headers)
         self.assertEqual(response.status_code, 400)
-        self.assertEqual(response.json(), {"errors": {"csv_file": ["[Row 1] reference: This field may not be blank."]}})
+        self.assertEqual(
+            response.json(), {"errors": {"csv_file": ["[Row 1] regime_reg_ref: This field may not be blank."]}}
```
Comment (currycoder) [inline api/external_data/tests/test_views.py:270]: It seems like the error triggered/error message has been changed here but `spire_entity_id` remains in the CSV?

**Review by currycoder — COMMENTED** [review]
lite-frontend has the following reference too; https://github.com/uktrade/lite-frontend/blob/dev/caseworker/external_data/example.csv Does that need adjusting as well?

**Comment by markj0hnst0n** [comment]
I think I've addressed the comments on this one @currycoder @depsiatwal

File: api/external_data/tests/test_views.py [hunk 3]
```diff
@@ -273,11 +262,13 @@ def test_update_success(self):
     def test_create_error_serializer_errors(self):
         url = reverse("external_data:denial-list")
         content = """reference,regime_reg_ref,name,address,notifying_government,country,item_list_codes,item_description,end_use,reason_for_refusal,spire_entity_id,entity_type
-        ,AB-CD-EF-000,Organisation Name,"1000 Street Name, City Name",Country Name,Country Name,0A00100,Medium Size Widget,Used in industry,Risk of outcome,123,end_user
+        DN2000/0000,,Organisation Name,"1000 Street Name, City Name",Country Name,Country Name,0A00100,Medium Size Widget,Used in industry,Risk of outcome,123,end_user
```
Comment (depsiatwal) [inline api/external_data/tests/test_views.py:265]: we still appears to have a reference here to spire_id

Comment (markj0hnst0n) [reply api/external_data/tests/test_views.py]: Addressed this issue @depsiatwal

**Review by Tllew — APPROVED** [review]
Look good to me

**Pull request merged**